### PR TITLE
added like and dislike videos, refractored video card component

### PR DESCRIPTION
**Pull request by JantuDeb**
# Liked Videos
- add a video to liked list
- remove a video from the liked list

# Video Card
- refactored video card, added modal inside the card itself.
- fixed alignment of videos 
image

### CODE REVIEW

File: src/components/shared/video-card/VideoCardActionMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const { removeFromPlaylist } = usePlaylist();
+
+  const addToPlayList = (id) => {
+    if (!playlistId) setShowModal((v) => !v);
+    else removeVideoHandler(videoId, playlistId);
+  };
+
+  const likeDislikeClickHandler = () => {
+    if (isInLiked) removeFromLikes(videoId);
+    else addToLikes(videoId);
```
Comment (rkapoor10) [inline src/components/shared/video-card/VideoCardActionMenu.jsx:22]: maybe ternay operator

File: src/components/shared/video-card/VideoCardActionMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  };
+
+  const removeVideoHandler = (videoId, playlistId) => {
+    if (videoId || playlistId) removeFromPlaylist({ playlistId, videoId });
```
Comment (rkapoor10) [inline src/components/shared/video-card/VideoCardActionMenu.jsx:26]: here also

**Review by rkapoor10 — APPROVED** [review]
The code is well structured and looks good to merge.

File: src/pages/home/Home.css [hunk 3]
```diff
@@ -2,7 +2,7 @@
   display: grid;
   grid-gap: 1rem;
   /* grid-template-columns: repeat(1, 1fr); */
-    grid-template-columns: repeat(auto-fit, minmax(16rem, 1fr));
+    grid-template-columns: repeat(auto-fill, minmax(16rem, 1fr));
```
Comment (sushil-mod) [inline src/pages/home/Home.css:5]: i liked how you used grid

**Review by sushil-mod — APPROVED** [review]
overall, nice clean code , i liked how you used your small component like a modal one . so much to learn and also your UI  is great

File: src/components/shared/video-card/VideoCardActionMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  };
+
+  return (
+    <>
+      <DropdownMenu>
+        <div
+          className="flex items-center gap-1 pointer"
+          onClick={() => console.log(videoId)}
```
Comment (Manju-sree) [inline src/components/shared/video-card/VideoCardActionMenu.jsx:34]: remove console.log if not necessary here

**Review by Manju-sree — APPROVED** [review]
Like the approach of writing code first using useState and converting them into useContext. always will get to learn new things from reviewing your code. please try to do code clean-up by removing all commented code. otherwise everything looks perfect in this PR.

File: src/components/shared/video-card/VideoCardActionMenu.jsx (hunk 4 above)
Comment (JantuDeb) [reply src/components/shared/video-card/VideoCardActionMenu.jsx:34]: Thank you, I will remove it.

**Pull request merged**